lib/index.js

```diff
@@ -3,9 +3,41 @@
 exports.register = (server, options, next) => {
 
   server.ext('onPostAuth', (request, reply) => {
-    request.payload.loaded = true;
+    request.plugins['hapi-rate-limit'] = { rate: null };
+
+    if (!(request.route.method === 'get' || request.route.method === 'post' || request.route.method === 'delete')) {
+      return reply.continue();
+    }
+
+    const rate = Object.assign({}, options.defaultRate);
+    const routeSettings = request.route.settings.plugins.rateLimit || null;
+
+    if (routeSettings) {
+      if (routeSettings.enabled === false) {
+        return reply.continue();
+      }
+
+      rate.limit = routeSettings.limit;
+      rate.window = routeSettings.window;
+    }
+
+    const key = `hapi-rate-limit:${request.route.method}:${request.route.path}:${request.auth.credentials.auth.api_key}`;
+
+    options.redisClient.multi();
+    options.redisClient.set(key, 0, 'EX', rate.window, 'NX'); // if key not found, insert key:0 with window seconds expiry
+    options.redisClient.incr(key);
+    return options.redisClient.exec()
+    .then((results) => {
+      rate.current = results[1]; // result of 2nd redis instruction
+
+      if (rate.current > rate.limit) {
+        return reply(options.overLimitError);
+      }
+
+      request.plugins['hapi-rate-limit'].rate = rate;
```

Comment:
even if an error was thrown because it was over the limit, it should still have the `rate` set, right?